renku/cli/run.py

```diff
@@ -25,35 +25,63 @@
 
 * arguments (flags),
 * string and integer options,
-* input files if linked to existing files in the repository,
-* output files if modified or created while running the command.
+* input files or directories if linked to existing paths in the repository,
+* output files or directories if modified or created while running the command.
 
-.. note:: If there were uncommitted changes then the command fails.
-   See :program:`git status` for details.
+.. note:: If there were uncommitted changes in the repository then the
+   ``renku run`` command fails. See :program:`git status` for details.
 
-Detecting input files
+.. warning:: Input and output paths can only be detected if they are passed as
+   arguments to ``renku run``.
+
+Detecting input paths
 ~~~~~~~~~~~~~~~~~~~~~
 
-An argument is identified as an input file only if its path matches an existing
-file in the repository. There might be several situations when the detection
-might not work as expected:
+Any path passed as an argument to ``renku run``, which was not changed during
+the execution is identified as an input path. The identification only works if
+the path associated with the argument matches an existing file or directory
+in the repository.
+
+The detection might not work as expected
+if:
+
+* a file is **modified** during the execution. In this case it will be stored
+  as an **output**;
+* a path is not passed as an argument to ``renku run``.
+
+Detecting output files and folders
```

Comment:
do you also want paths here instead of files and folders?
```suggestion
Detecting output paths
```